### ISSUE TITLE
Computer requirements

### ISSUE THREAD

**Issue by bbugert** [FIRST-Tech-Challenge/scorekeeper#403]
Can't find in any manual.
What are the computer requirements to run the web based software, other than what is on page 14 of the manual for Windows? ie. CPU, RAM, Hard Drive space, ect...

**Comment by GamingWithaAlexYouTube** [comment]
What specs does your computer have?

**Comment by bbugert** [comment]
Need to know what
   -CPU speed is best I5, I6, I7 or I8 (especially for a Dual Division Regional Championship)
   -RAM 8G or 16G
   -Hard Drive space 500G or 1T

**Comment by GamingWithaAlexYouTube** [comment]
the worst computer i have is a CPU Intel core I5 and it has 8GB of Ram and it has a 237GB hard drive. so if you have at least that or more you will be way more than fine

**Comment by jvens** [comment]
@bbugert, keep in mind that I am not on the dev team for this project, and therefore take what I say with a grain of salt.

In my experience this system is very resource light.  You are not going to have any problems with any relatively modern (say last 5 years) computer.  You only need around 100MB of hard drive space, a few hundred MB of RAM, and any processor and it will run just fine.  I have even run the system off of a Raspberry Pi 3 and while I wouldn't recommend that for a dual division regional championship, it still works just fine.

What you do need, especially for larger events, is a decent WiFi router if you plan to do live scoring.  With 4 fields each with 2 refs and then the displays you can easily end up with 20+ devices at the same time.  Also, at an event with 48 teams and 4 fields it can easily be over 100 feet between the outside fields, and with all the robots competing for WiFi space it is a noisy environment.  I recommend using 5GHz if possible, and if you have to use 2.4GHz then put the score system on whatever channel you are not using for robots (remember to only use channels 1, 6, and 11).  Also, with so many concurrent connections to the score system at a large event like that I would use a computer with a gigabit ethernet connection (all reasonably modern computers have this).

tl;dr any old computer you have laying around will almost certainly run the system just fine.

**Comment by ftctechnh** [comment]
@bbugert 
Thanks bbugert.  This is a good question!  We're revising the Scorekeeper manual to include suggestions on the hardware requirements.  Here's a preview of the recommended minimum hardware:

## Minimum Hardware Configuration
For most events, the hardware requirements to run the FTC Live scoring software are very modest.
### Scoring Server Minimum Requirements (Windows/Mac/Linux)
* Intel Core i5 Processor
* 4 GB of RAM
* 300 MB of available disk space
* CAT 5 Ethernet adapter (USB connected would be OK if built-in not available)
* 802.11 WiFi adapter

Note that if you are running a large event, such as a dual division event, a higher performing computer might be warranted.

### Network Requirements
The selection and configuration of the network hardware can be more critical than the selection of the server hardware to the success of your live scoring event.  
* Use a dedicated (i.e., not shared with other users or applications), local network to connect your scoring devices.
* To avoid latency issues, the scoring server should be local to your event (and not run through a remote Internet/Cloud connection).
* Use a wired connection whenever possible (with the exception of the realtime scoretracking tablets, which are typically wireless devices).
* If you need to use a wireless connection, work with your FTA and venue IT staff to make sure you select a wireless channel that does not interfere with the competition robots.
* Check with your venue’s IT staff to see if a WiFi Blocker is present.  If one is present, you will not be able to use wireless scoretracking devices unless the staff disables this function.
* Use a modern (made within the past 3 years) switch/access point that is powerful enough to accommodate the number of devices needed at your event
     - FIRST has received credible reports of older, “underpowered” switch/access points being unable to support a large enough number of devices operating on the scoring system network.
     - Use an appropriately sized network device that will be able to accommodate all your scoring-related devices simultaneously

**Comment by jvens** [comment]
One small comment Tom, if the venue has a WiFi Blocker that then keeping the scoring tablets connected is going to be the least of your problem.  I would recommend changing this bullet point to "If one is present, you will need to work with the IT staff to disable this function." as this can effect robot WiFi Direct connections in addition to the score system.
